src/extractor/extraction_containers.cpp

```diff
@@ -53,6 +112,12 @@ ExtractionContainers::ExtractionContainers()
     name_lengths.push_back(0);
     name_lengths.push_back(0);
 
+    // Insert three zero offsets for empty strings and the total length
+    name_offsets.push_back(0);
+    name_offsets.push_back(0);
+    name_offsets.push_back(0);
+    name_offsets.push_back(0);
+
```

Comment:
No, the last offset is always the total length of the character data array and at the moment it is zero. Offsets are added after pushing data into the data  array.